### PR TITLE
Feat/movie service

### PR DESCRIPTION
**Pull request by bhavindesai20**
Please review my code and provide your comments.

### CODE REVIEW

File: src/com/atlas/controller/TitleController.java [hunk 1]
```diff
@@ -18,18 +23,54 @@
...
+	public Title getTitleById(@PathVariable("id") int id) throws MovieNotFound{
+		return this.titleService.getTitleById(id);
+	}
 
 	@RequestMapping(method = RequestMethod.POST, produces=MediaType.APPLICATION_JSON_VALUE,
 			consumes=MediaType.APPLICATION_JSON_VALUE)
-	public Title addTitle(@RequestBody Title t) {
-		this.titleService.addTitle(t);
-		return t;
+	public Title addTitle(@RequestBody Title t) throws MovieBadRequest{
```
Comment (surya1024) [inline src/com/atlas/controller/TitleController.java:41]: I dont think there is a need to return the object, 200 Status code is good enough

File: src/com/atlas/entity/Rating.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id")
+	private int id;
+
+	@Column(name = "rating")
+	private int rating;
+	
+	@JsonIgnore
```
Comment (surya1024) [inline src/com/atlas/entity/Rating.java:34]: there is no need to use JsonIgnore, we may have to send title to front end.

File: src/com/atlas/entity/Rating.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	@ManyToOne
+	@JoinColumn(name = "title_Id", nullable = false)
+	private Title title;
+
+	@JsonIgnore
```
Comment (surya1024) [inline src/com/atlas/entity/Rating.java:39]: there is no need to use JsonIgnore, we may have to send user to front end.

File: src/com/atlas/controller/TitleController.java (hunk 1 above)
Comment (bhavindesai20) [inline src/com/atlas/controller/TitleController.java:41]: I am returning this object as we need to confirm on UI side too that we have this entire object in list. if we just return HttpStatus, we can't show them in grid instantly after add.

File: src/com/atlas/entity/Rating.java (hunk 2 above)
Comment (bhavindesai20) [inline src/com/atlas/entity/Rating.java:34]: @JsonIgnore is used to not sending entire title with each rating.

File: src/com/atlas/entity/Rating.java (hunk 3 above)
Comment (bhavindesai20) [inline src/com/atlas/entity/Rating.java:39]: @JsonIgnore is used to not sending entire user detail with each rating.

**Comment by surya1024** [comment]
:+1:

**Pull request merged**